### PR TITLE
Redial Ethereum connection on every call, fixes #1160

### LINKED ISSUE #1160: ethereum connector recovery from a connection failure
**Describe the bug**

The connection object (the client) is 'dialed if needed' only if the object is nil - there is no logic to refresh it if it fails during runtime or if the node changes (dns change) for example.

it is possible that a flow exists where it does recover, but it should be refactored (the dial logic) to be more robust.

**Steps To Reproduce**

1. run the system
2. kill the ethereum node
3. it will not try to reconnect properly

**Expected behavior**

an ability to recover / restart the connection, either manually or periodically.

**Logs**
n/a
**Screenshots**
n/a
**Environment:**
n/a (all)
**Additional context**
n/a

### PR DESCRIPTION
**Pull request by netoneko**
## Description

Fixes #1160 

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/orbs-network/orbs-contributor-guide/blob/master/CONTRIBUTING.md) doc
- [ ] I have opened an issue and discussed the solution suggested in this PR
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/crosschainconnector/ethereum/adapter/rpc_connection.go [hunk 1]
```diff
@@ -33,36 +27,17 @@ func NewEthereumRpcConnection(config ethereumAdapterConfig, logger log.Logger) *
 	}
 	rpc.logger = logger.WithTags(log.String("adapter", "ethereum"))
 	rpc.getContractCaller = func() (caller EthereumCaller, e error) {
-		return rpc.dialIfNeededAndReturnClient()
+		return rpc.dial()
```
Comment (electricmonk) [inline services/crosschainconnector/ethereum/adapter/rpc_connection.go]: Yeah, I was wondering why we were reusing the same connection. Good catch.

**Comment by electricmonk** [comment]
@netoneko @jlevison I'm approving this - but:
Please make sure to add a benchmark (gobench) that measures call latency with cached vs recreated connections, to make sure that the redial operation doesn't add significant latency to the call. Ethereum calls might occur at a fast pace, so if there's significant latency incurred, we might wanna opt for a stabler cached connection solution.

**Comment by netoneko** [comment]
@electricmonk this concern will only arise if someone will actually start using the system to communicate with Ethereum, which we haven't tried yet even in tests

**Comment by electricmonk** [comment]
@netoneko we already do that from the Elections contract. Please add the benchmark now. Otherwise, no one will ever remember to even check the Ethereum connector later.

**Pull request merged**